beacon_node/execution_layer/src/lib.rs

```diff
@@ -606,132 +652,193 @@ impl<T: EthSpec> ExecutionLayer<T> {
                         "Requesting blinded header from connected builder";
                         "slot" => ?slot,
                         "pubkey" => ?pubkey,
-                        "parent_hash" => ?parent_hash,
+                        "parent_hash" => %parent_hash,
                     );
-                    let (relay_result, local_result) = tokio::join!(
-                        builder.get_builder_header::<T, Payload>(slot, parent_hash, &pubkey),
-                        self.get_full_payload_caching(
-                            parent_hash,
-                            timestamp,
-                            prev_randao,
-                            suggested_fee_recipient,
-                            forkchoice_update_params,
-                        )
+
+                    // Wait for the builder *and* local EL to produce a payload (or return an error).
+                    let ((relay_result, relay_duration), (local_result, local_duration)) = tokio::join!(
+                        timed_future(metrics::GET_BLINDED_PAYLOAD_BUILDER, async {
+                            builder
+                                .get_builder_header::<T, Payload>(slot, parent_hash, &pubkey)
+                                .await
+                        }),
+                        timed_future(metrics::GET_BLINDED_PAYLOAD_LOCAL, async {
+                            self.get_full_payload_caching::<Payload>(
+                                parent_hash,
+                                timestamp,
+                                prev_randao,
+                                suggested_fee_recipient,
+                                forkchoice_update_params,
+                            )
+                            .await
+                        })
+                    );
+
+                    info!(
+                        self.log(),
+                        "Requested blinded execution payload";
+                        "relay_fee_recipient" => match &relay_result {
```

Comment:
We may want to use the debug impl for the fee recipients too

Reply:
Good call, also fixed in e8bd2be26